### PR TITLE
Bringing TS flow to same steps as curl-based flow

### PR DESCRIPTION
**Pull request by julioalex-rezende**
Adding new validations and checks to Typescript flow to reflect what is done with curl commands.

Some light changes that don't affect the original behavior, such as :
- validations' ordering to better adapt to TS logic, 
- Full report displayed for all members, as TS flow gives a nicer presentation view with tables.

### CODE REVIEW

File: data-reconciliation-app/test/e2e-test/src/api.ts [hunk 1]
```diff
@@ -48,9 +45,39 @@ export default class Api {
...
         }
 
-        console.log(`✅ [PASS] [${result.status} : ${result.statusText}] - ${member.name}\n`);
+        console.log(`✅ [PASS] [${result.status} : ${result.statusText}] - ${member.name}`);
         console.table(result.data.content);
 
         return result.data.content;
     }
+
+    public static async checkPostResponseResult(apiUrl: string, member: DemoMemberProps, expectedStatus: number, testMessage: string) {
```
Comment (macromania) [inline data-reconciliation-app/test/e2e-test/src/api.ts:54]: Would it be a better name if we call it and merge the two methods into one?

Also, I think `API` class is mainly describing and running the endpoints. This method is more like a validator.

```typescript
export interface ValidationProps {
   url: string;
   method: 'GET' | 'POST';
   member: DemoMemberProps;
   expectedStatus: number;
   testMessage: string;
}

export class Validator {
   public static async validateRequest(props: ValidationProps) {
      //
      const status = await axios({
         method: props.method,
         url: props.url,
         data: props.member.data,
         httpsAgent: props.member.httpsAgent,
         validateStatus: function (status) {
            return status < 500; // Resolve only if the status code is less than 500
        },
       });

     if (result.status !== expectedStatus) {
         throw new Error(`🛑 [TEST FAILURE]: ${member.name} - ${testMessage}: ${expectedStatus} expected, but got ${result.status}`);
     }

     console.log(`✅ [PASS] ${member.name} - ${testMessage}`);
   }
}
```

Comment (julioalex-rezende) [reply data-reconciliation-app/test/e2e-test/src/api.ts]: done

**Review by Aymalla — APPROVED** [review]
LGTM

**Review by mkcomer — APPROVED** [review]
LGTM

**Pull request merged**